pkgs/development/python-modules/nlpo3/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  buildPythonPackage,
+  rustPlatform,
+  fetchFromGitHub,
+  pythonAtLeast,
+  rustc,
+  cargo,
+  setuptools,
+  setuptools-rust,
+  wheel,
+}:
+
+buildPythonPackage rec {
+  pname = "nlpo3";
+  version = "1.3.0";
+  pyproject = true;
+
+  disabled = pythonAtLeast "3.12";
+
+  src = fetchFromGitHub {
+    owner = "PyThaiNLP";
+    repo = "nlpo3";
+    rev = "refs/tags/nlpo3-python-v${version}";
+    hash = "sha256-A1q6iieoPuRFTnyzg9y3Tp/rMx1c8nbcd1f8aaqonf4=";
+  };
+
+  sourceRoot = "${src.name}/nlpo3-python";
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src sourceRoot;
+    hash = "sha256-OWXD5t4Jg9/GLgwIvM1j4Ns5oFFaHWlZbGMh3cBsIQw=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    rustc
+    cargo
+  ];
+
```

Comment:
add `buildInputs = lib.optionals stdenv.isDarwin [ libiconv ];`, this will fix darwin build